### ISSUE TITLE
jooby2.x io.jooby.Usage: Router not found 

### ISSUE THREAD

**Issue by j2eekingcn** [jooby-project/jooby#1692]
hello,
I have set according to the documentation

	<plugin>
				<artifactId>maven-compiler-plugin</artifactId>
				<version>3.8.1</version>
				<configuration>
    				<forceJavacCompilerUse>true</forceJavacCompilerUse>
					
					<annotationProcessorPaths>
						<path>
						    <groupId>org.projectlombok</groupId>
						    <artifactId>lombok</artifactId>
						    <version>1.18.12</version>
						</path>
						<path>
							<groupId>io.jooby</groupId>
							<artifactId>jooby-apt</artifactId>
							<version>2.8.1</version>
						</path>
					</annotationProcessorPaths>
				</configuration>
			</plugin>
			<plugin>
				<groupId>io.jooby</groupId>
				<artifactId>jooby-maven-plugin</artifactId>
				<version>2.8.1</version>
				<configuration>
					<mainClass>${application.class}</mainClass>
					<restartExtensions>conf,properties,class</restartExtensions>
					<compileExtensions>java,kt</compileExtensions>
					<port>8080</port>
				</configuration>
			</plugin>

But after I modified the code
Console output error

io.jooby.Usage[m: [1;31mRouter not found: `cn.ef.modules.sys.action.DownloadAct`. Make sure Jooby annotation processor is configured properly.
For more details, please visit: https://jooby.io/usage#annotation-processing-tool-router-not-found[m
    [1mat[m io.jooby.Usage.apt ([1mUsage.java:62[m)
    [1mat[m io.jooby.Usage.mvcRouterNotFound ([1mUsage.java:41[m)
    [1mat[m io.jooby.Jooby.lambda$null$3 ([1mJooby.java:323[m)
    [1mat[m java.util.Optional.orElseThrow ([1mOptional.java:290[m)
    [1mat[m io.jooby.Jooby.lambda$mvc$4 ([1mJooby.java:323[m)
    [1mat[m java.util.Optional.orElseGet ([1mOptional.java:267[m)
    [1mat[m io.jooby.Jooby.mvc ([1mJooby.java:320[m)
    [1mat[m io.jooby.Jooby.mvc ([1mJooby.java:310[m)

**Comment by j2eekingcn** [comment]
[DEBUG] ENTRY_MODIFY [C:\Users\Administrator\Desktop\jooby\ef-jooby2\ef-site-parent\ef-site\target\classes\cn\br\ef\modules\sys\action\SimpleAct$Module.class]
[DEBUG] ENTRY_MODIFY [C:\Users\Administrator\Desktop\jooby\ef-jooby2\ef-site-parent\ef-site\target\classes\META-INF\services\io.jooby.MvcFactory]
[DEBUG] Ignoring file change: C:\Users\Administrator\Desktop\jooby\ef-jooby2\ef-site-parent\ef-site\target\classes\META-INF\services\io.jooby.MvcFactory
[DEBUG] ENTRY_MODIFY [C:\Users\Administrator\Desktop\jooby\ef-jooby2\ef-site-parent\ef-site\target\classes\cn\br\ef\modules\sys\action\SimpleAct.class]
[DEBUG] Restarting application on file change: C:\Users\Administrator\Desktop\jooby\ef-jooby2\ef-site-parent\ef-site\target\classes\cn\br\ef\modules\sys\action\SimpleAct.class
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[ERROR] execution of cn.br.Boot resulted in exception
io.jooby.Usage: Router not found: `cn.br.ef.modules.sys.action.DownloadAct`. Make sure Jooby annotation processor is configured properly.
For more details, please visit: https://jooby.io/usage#annotation-processing-tool-router-not-found
    at io.jooby.Usage.apt (Usage.java:62)
    at io.jooby.Usage.mvcRouterNotFound (Usage.java:41)
    at io.jooby.Jooby.lambda$null$3 (Jooby.java:323)
    at java.util.Optional.orElseThrow (Optional.java:290)
    at io.jooby.Jooby.lambda$mvc$4 (Jooby.java:323)
    at java.util.Optional.orElseGet (Optional.java:267)
    at io.jooby.Jooby.mvc (Jooby.java:320)
    at io.jooby.Jooby.mvc (Jooby.java:310)
    at cn.br.ef.MvcApiMain.addRoutes (MvcApiMain.java:230)
    at cn.br.ef.MvcApiMain.init (MvcApiMain.java:61)
    at cn.br.ef.MvcApiMain.<init> (MvcApiMain.java:147)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0 (Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:62)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance (Constructor.java:423)
    at io.jooby.Jooby.lambda$null$10 (Jooby.java:1068)
    at io.jooby.SneakyThrows$Function.apply (SneakyThrows.java:521)
    at java.util.Optional.map (Optional.java:215)
    at io.jooby.Jooby.lambda$reflectionProvider$12 (Jooby.java:1068)
    at io.jooby.Jooby.createApp (Jooby.java:964)
    at io.jooby.Jooby.runApp (Jooby.java:936)
    at io.jooby.Jooby.runApp (Jooby.java:876)
    at cn.br.Boot.main (Boot.java:16)

**Comment by jknack** [comment]
Hi,

Can you provide a simple test project to reproduce it?

Thanks.

**Comment by j2eekingcn** [comment]
It ’s strange after I simplified the project
Modify the code is normal
I guess there must be something wrong there
I will look for it myself

**Comment by jknack** [comment]
Cool. Let me know what you find

**Comment by j2eekingcn** [comment]
Initially found the problem
When i put in this controller

package cn.br.ef.modules.sys.action;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import cn.br.common.exception.UploadDownloadException;
import io.jooby.AttachedFile;
import io.jooby.Context;
import io.jooby.StatusCode;
import io.jooby.annotations.GET;
import io.jooby.annotations.Path;

import java.io.File;
import java.net.URL;
import java.nio.file.Paths;

import static java.util.Objects.nonNull;

@Path("/mvc")
public class DownloadAct {

	private static final Logger log = LoggerFactory.getLogger(DownloadAct.class);
	private static final String JOOBY_LOGO = "logo_jooby.png";
	private static final String ERROR = "MVC Download error: %s.";

	@GET
	@Path("/download")
	public AttachedFile accept(Context ctx) {
		try {
//			final URL logo = this.getClass().getClassLoader().getResource(JOOBY_LOGO);
//			assert nonNull(logo);
//			final File file = new File(logo.toURI());

			java.nio.file.Path source = Paths.get("logo.png");
//			log.info("MVC Downloading file with size: " + source.length());
//			ctx.setResponseCode(StatusCode.OK).send(file)
//			response.status(200).download(file);
			return new AttachedFile(source);
		} catch (Throwable e) {
			final String error = String.format(ERROR, e.getMessage());
			throw new UploadDownloadException(error, e);
		}
	}
}


Modify the code, hot loading is useless, it will throw the above exception

**Comment by jknack** [comment]
Should be something else... I used your code, edit several times, no error.

```java
import io.jooby.AttachedFile;
import io.jooby.Context;
import io.jooby.annotations.GET;
import io.jooby.annotations.Path;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.File;
import java.net.URL;
import java.nio.file.Paths;

import static java.util.Objects.nonNull;

@Path("/mvc")
public class DownloadAct {

  private static final Logger log = LoggerFactory.getLogger(DownloadAct.class);

  private static final String JOOBY_LOGO = "logo_jooby.png";
  private static final String ERROR = "MVC Download error: %s.";

  @GET
  @Path("/download")
  public AttachedFile accept(Context ctx) {
    try {
      ClassLoader classLoader = this.getClass().getClassLoader();
      final URL logo = classLoader.getResource(JOOBY_LOGO);
      assert nonNull(logo);
      final File file = new File(logo.toURI());

      java.nio.file.Path source = Paths.get("logo.png");
      log.info("MVC Downloading file with size: " + file.length());
      return new AttachedFile(source);
    } catch (Throwable e) {
      final String error = String.format(ERROR, e.getMessage());
      throw new UploadDownloadException(error, e);
    }
  }
}
```

**Comment by j2eekingcn** [comment]
i am using eclipse and lombok plugin
I delete DownloadAct then Normal
put it in project then Error

**Comment by j2eekingcn** [comment]
I uploaded the project to 
https://github.com/j2eekingcn/ef-demo

**Comment by j2eekingcn** [comment]
Easy to reproduce
Annotate DownloadAct class
mvn jooby: run is normal
Uncomment the DownloadAct class
Modify SimpleAct
ctx.setResponseType (MediaType.TEXT) .send ("hello World 232423433333344444!");
Ctx.setResponseType (MediaType.TEXT) .send ("hello World!");

Spring Tool Suite 4.6.1
lombok plugin

**Comment by jknack** [comment]
It works. But I have to fix some compilation problems and removed the HikariModule(don't have a database).

Try removing the `jooby-run` dependency from your path. This is not required and may cause some troubles.

If that doesn't fix it could be. something related to the Eclipse incremental compiler and. then way it synchronize the changes vs when jooby run reload the app...but not 100% sure (I'm an Intellij user)

**Comment by j2eekingcn** [comment]
Thank you very much, I have deleted jooby-run and HikariModule according to your guidance, the problem still exists, I will try idea tomorrow

**Comment by jknack** [comment]
sorry dude, but yea try IDEA and let me know

**Comment by j2eekingcn** [comment]
**The following idea also has this problem
Is there a problem with the method I use**

[ERROR] execution of cn.br.Boot resulted in exception
io.jooby.Usage: Router not found: `cn.br.ef.modules.sys.action.DownloadAct`. Make sure Jooby annotation processor is configured properly.
For more details, please visit: https://jooby.io/usage#annotation-processing-tool-router-not-found
    at io.jooby.Usage.apt (Usage.java:62)
    at io.jooby.Usage.mvcRouterNotFound (Usage.java:41)
    at io.jooby.Jooby.lambda$null$3 (Jooby.java:323)
    at java.util.Optional.orElseThrow (Optional.java:290)
    at io.jooby.Jooby.lambda$mvc$4 (Jooby.java:323)
    at java.util.Optional.orElseGet (Optional.java:267)
    at io.jooby.Jooby.mvc (Jooby.java:320)
    at io.jooby.Jooby.mvc (Jooby.java:310)
    at cn.br.ef.MvcApiMain.addRoutes (MvcApiMain.java:69)
    at cn.br.ef.MvcApiMain.init (MvcApiMain.java:36)
    at cn.br.ef.MvcApiMain.<init> (MvcApiMain.java:44)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0 (Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:62)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance (Constructor.java:423)
    at io.jooby.Jooby.lambda$null$10 (Jooby.java:1068)
    at io.jooby.SneakyThrows$Function.apply (SneakyThrows.java:521)
    at java.util.Optional.map (Optional.java:215)
    at io.jooby.Jooby.lambda$reflectionProvider$12 (Jooby.java:1068)
    at io.jooby.Jooby.createApp (Jooby.java:964)
    at io.jooby.Jooby.runApp (Jooby.java:936)
    at io.jooby.Jooby.runApp (Jooby.java:876)
    at cn.br.Boot.main (Boot.java:16)


**Start project mvn jooby: run
Visit http: // localhost: 8080 / mvc normal
Then uncomment all DownloadAct
Click build project
Then the exception appeared**

**Comment by jknack** [comment]
Did you click 'Build Project' on IDEA?

You work with Idea by saving the file. Jooby run detects the changes in the .java files and ask Maven to compile them. Make sure Eclipse classpath metadata isn't present (.classpath file), otherwise jooby run won't fire a maven compilation

**Comment by jknack** [comment]
Here is what I did:

- Download Eclipse.
- Import project as Maven Project
- On terminal: mvn clean jooby:run
- Open a browser at: http://localhost:8080/mvc
- Comment/Uncomment lines on DownloadAct
- Hit Save (Command + S)

Jooby restart without issues:

```
```

**Comment by j2eekingcn** [comment]
I feel like it may be an environmental issue?
My environment is win10 + Spring Tools 4.6.1 + Maven 3.6.1

**Comment by jknack** [comment]
This is fixed in #1767
